### ISSUE TITLE
Gretty 2.3.0 JettyScannerManager error

### ISSUE THREAD

**Issue by acestars** [gretty-gradle-plugin/gretty#99]
This is the configuration:
```
managedClassReload = false
servletContainer = 'tomcat9'
scanInterval=1
scanDir "${projectDir}/src/main"
fastReload "${projectDir}/WebContent"
```
every I changed the content of WebContent the JettyScannerManager throw error
```
org.akhikhl.gretty.JettyScannerManager$1@4f2ec1aa failed on '[path_of_directory, path_of_directory/filename]
java.lang.IllegalAccessError: tried to access method org.akhikhl.gretty.scanner.BaseScannerManager.scanFilesChanged(Ljava/util/Collection;)V from class org.akhikhl.gretty.JettyScannerManager$1
	at org.akhikhl.gretty.JettyScannerManager$1.filesChanged(JettyScannerManager.groovy:33)
	at org.eclipse.jetty.util.Scanner.reportBulkChanges(Scanner.java:691)
	at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:551)
	at org.eclipse.jetty.util.Scanner.scan(Scanner.java:403)
	at org.eclipse.jetty.util.Scanner$1.run(Scanner.java:353)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
```

gretty 2.2.0 works properly, no error

**Comment by javabrett** [comment]
Thanks for the report.  Which JDK are you using, and can you please provide a minimal example project.

**Comment by boris-petrov** [comment]
I just noticed the same issue. Could it be related to https://github.com/gretty-gradle-plugin/gretty/issues/95? The new Groovy perhaps outputs some broken bytecode because of the compile error?

**Comment by javabrett** [comment]
@boris-petrov an excellent theory.  I have some local commits with a fix in the works for that.  If we can document a reproducible case for this, it can be tested (might just rely on `scanDir` or `fastReload`).

**Comment by acestars** [comment]
I already tried jdk 8 and 11. Maybe I will provide the sample project on 
monday.

**Comment by javabrett** [comment]
I can reproduce this also.  The problem call is at:

https://github.com/gretty-gradle-plugin/gretty/blob/41d081cf64873ad834124de6074e4b7c701eb394/libs/gretty/src/main/groovy/org/akhikhl/gretty/JettyScannerManager.groovy#L33

I think this is caused by Groovy bug [Statically compiled calls to protected methods of an outerclass' superclass result in IllegalAccessErrors](https://issues.apache.org/jira/browse/GROOVY-7862) (fixed in Groovy 2.5), together with 3cb25470694d86898f8e72df9da2b05c6766d3db which added `@CompileStatic` to as part of #46.

https://github.com/gretty-gradle-plugin/gretty/blob/41d081cf64873ad834124de6074e4b7c701eb394/libs/gretty/src/main/groovy/org/akhikhl/gretty/scanner/BaseScannerManager.groovy#L15

We can either remove that one `@CompileStatic` or compensate for the bug.
